src/platform/intel/ace/lib/pm_runtime.c

```diff
@@ -59,6 +60,40 @@ const struct pm_state_info *pm_policy_next_state(uint8_t cpu, int32_t ticks)
 }
 #endif /* CONFIG_PM_POLICY_CUSTOM */
 
+static inline void ace_pm_runtime_allow_pgcg(bool enable)
+{
+	if (enable) {
+		pm_policy_state_lock_put(PM_STATE_RUNTIME_IDLE, PM_ALL_SUBSTATES);
+	} else {
+		pm_policy_state_lock_get(PM_STATE_RUNTIME_IDLE, PM_ALL_SUBSTATES);
+		/* Disable power gating when preventing */
+		DFDSPBRCP.bootctl[PLATFORM_PRIMARY_CORE_ID].bctl |=
+			DFDSPBRCP_BCTL_WAITIPCG | DFDSPBRCP_BCTL_WAITIPPG;
+	}
```

Comment:
I'm not familiar with Zephyr PM API, are we safe here from `put()` or `get()` being unbalanced? I.e. can this function be called multiple times with the same value of `enable`?

Reply:
Functions can be called multiple times but to balance the count of it is on the application side.